sceptre/connection_manager.py

```diff
@@ -170,13 +179,36 @@ def _get_client(self, service):
         :rtype: boto3.client.Client
         """
         with self._client_lock:
+            if self._is_session_expired():
+                self.clients[service] = None
+
             if self.clients.get(service) is None:
                 self.logger.debug(
                     "No %s client found, creating one...", service
                 )
                 self.clients[service] = self.boto_session.client(service)
             return self.clients[service]
 
+    def _is_session_expired(self):
+        """
+        Validates the boto session to make sure it is still valid and returns
+        false if it has expired
+
+        :returns: session expired boolean
+        :rtype: boolean
+        """
+
+        if (
+            self.iam_role and
+            self._boto_session_expiration >= datetime.now(tz.tzutc())
```

Comment:
can we set a variable `expired = self._boto_session_expiration >= datetime.now(tz.tzutc())`, then put if statement on one line ` if self.iam_role and expired:`